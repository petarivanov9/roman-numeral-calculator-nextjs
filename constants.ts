export const messages = {
  msgRequiredInteger: 'Must enter an integer!',
  msgRequiredPosInteger: 'Must enter a positive integer!',
};
