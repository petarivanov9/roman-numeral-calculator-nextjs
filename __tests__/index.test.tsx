import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Home, { testIds } from '@/pages/index';
import { messages } from '../constants';

const { msgRequiredInteger, msgRequiredPosInteger } = messages;

describe('Home', () => {
  it('should match snapshot', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  it('should convert `317` to `CCCXVII`', async () => {
    render(<Home />);

    await userEvent.type(screen.getByTestId(testIds.naturalInput), '317');

    await userEvent.click(screen.getByTestId(testIds.submit));

    expect(await screen.findByText(/CCCXVII/)).toBeInTheDocument();
  });

  it('should display `Must enter an integer!` when converting `random`', async () => {
    render(<Home />);

    await userEvent.type(screen.getByTestId(testIds.naturalInput), 'random');

    await userEvent.click(screen.getByTestId(testIds.submit));

    expect(await screen.findByText(msgRequiredInteger)).toBeInTheDocument();
  });

  it('should display `Must enter a positive integer!` when converting `-12`', async () => {
    render(<Home />);

    await userEvent.type(screen.getByTestId(testIds.naturalInput), '-12');

    await userEvent.click(screen.getByTestId(testIds.submit));

    expect(await screen.findByText(msgRequiredPosInteger)).toBeInTheDocument();
  });
});
