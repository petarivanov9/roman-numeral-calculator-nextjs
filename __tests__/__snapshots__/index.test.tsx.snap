// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <div
    class="h-full bg-blue-800"
  >
    <div
      class="max-w-full h-full px-4"
    >
      <div
        class="flex flex-wrap h-full justify-center"
      >
        <div
          class="flex flex-row items-center xl:w-4/12 lg:w-6/12 sm:w-8/12 w-full"
        >
          <div
            class="w-full"
          >
            <h1
              class="mb-12 text-white text-center"
            >
              Roman Calculator
            </h1>
            <main
              class="bg-white rounded-lg p-4"
            >
              <form>
                <div
                  class="md:p-4"
                >
                  <label
                    class="block text-sm font-medium text-gray-700"
                  >
                    Enter natural number:
                  </label>
                  <input
                    class="placeholder-gray-400 leading-6 w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm transition ease-in-out duration-100 focus:ring-indigo-500 focus:border-indigo-500 focus:outline-none focus:ring-1 focus:ring-offset-0"
                    id="natural"
                    name="natural"
                    placeholder="Ex: 123"
                    type="text"
                  />
                </div>
                <div
                  class="md:p-4"
                >
                  <button
                    class="border border-transparent rounded text-white ring-gray-200 m-0.5 px-2.5 py-1.5 transition-all ease-in duration-75 hover:bg-gray-100 hover:shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 bg-indigo-500 text-white ring-indigo-500 hover:bg-indigo-600 w-full"
                    type="submit"
                  >
                    Convert
                  </button>
                </div>
              </form>
              <div
                class="md:p-4"
              >
                Result: 
                <span>
                  N/A
                </span>
              </div>
            </main>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
